couchdb/server/helpers.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+#
+from inspect import getsource
+from textwrap import dedent
+from types import FunctionType
+
+from couchdb import util
+
+
+def maybe_extract_source(fun):
+    if isinstance(fun, FunctionType):
+        return dedent(getsource(fun))
+    elif isinstance(fun, util.strbase):
+        return fun
+    raise TypeError('Function object or source string expected, got %r' % fun)
```

Comment:
Yes, I know. Same story with lambdas. Basically, both are not suitable for any kind of design function, so no reason to care about.